### PR TITLE
nextcloud29: init at 29.0.0, nextcloud27: 27.1.8 -> 27.1.9, nextcloud28: 28.0.4 -> 28.0.5

### PR DESCRIPTION
**Pull request by RaitoBezarius**
## Description of changes

https://nextcloud.com/blog/nextcloud-hub8/

**Untested**
For NC maintainers: feel free to take over if I become unresponsive.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@Ma27 we now drop 26, or don't we?

**Comment by Ma27** [comment]
Yes. Can file a follow-up tomorrowish for that.

**Comment by Tom-Hubrecht** [comment]
Tested on my infra, it seems to work fine

**Comment by Ma27** [comment]
Pushed a small fixup, updated other nextclouds & removed nc26. Will need a bit more testing, but then it's ready from my PoV.

Another review would be greatly appreciated though.

**Comment by Ma27** [comment]
btw @RaitoBezarius do the changes I pushed look reasonable to you?

**Comment by Ma27** [comment]
I'm still having a few app-related issues that I'm figuring out today/tomorrow. Stand by.

**Comment by SuperSandro2000** [comment]
I am not planning to upgrade immediately to not catch up the release bugs.

**Review by bachp — APPROVED** [review]
27.1.9 is working

**Review by Ma27 — APPROVED** [review]
Approval feels kinda weird, since I'm a coauthor.
Anyways, will leave this open until ~tomorrow to give others a chance to review as well.

**Review by onny — APPROVED** [review]

**Comment by Ma27** [comment]
Ran another update since that fixes the broken app in question. Not sure if/how we want to do patches generally for apps, so I dropped that portion entirely and pushed another update round.

**Review by Shawn8901 — APPROVED** [review]
roughly checked 28.0.5 update and then upgraded that instance to 29. both looked okay on my setup

File: nixos/tests/nextcloud/default.nix [hunk 1]
```diff
@@ -22,4 +22,4 @@ foldl
     };
   })
 { }
-  [ 26 27 28 ]
+  [ 26 27 28 29 ]
```
Comment (SuperSandro2000) [inline nixos/tests/nextcloud/default.nix:25]:
```suggestion
  [ 27 28 29 ]
```

Comment (Ma27) [reply nixos/tests/nextcloud/default.nix]: Good catch, fixed.

File: nixos/modules/services/web-apps/nextcloud.nix [hunk 2]
```diff
@@ -819,7 +819,8 @@ in {
         ++ (optional (versionOlder cfg.package.version "25") (upgradeWarning 24 "22.11"))
         ++ (optional (versionOlder cfg.package.version "26") (upgradeWarning 25 "23.05"))
         ++ (optional (versionOlder cfg.package.version "27") (upgradeWarning 26 "23.11"))
-        ++ (optional (versionOlder cfg.package.version "28") (upgradeWarning 27 "24.05"));
+        ++ (optional (versionOlder cfg.package.version "28") (upgradeWarning 27 "24.05"))
+        ++ (optional (versionOlder cfg.package.version "29") (upgradeWarning 28 "24.11"));
```
Comment (Shawn8901) [inline nixos/modules/services/web-apps/nextcloud.nix:823]: As i am currently trying to upgrade a nextcloud server from 23.11 to 24.05 that warning feels a bit off as the defaults and the generated warning messages to not fit.

On `stateVersion` 24.05 we set a nextcloud29, but do not warn for having nextcloud29 available in 24.05 as default.

Also i have the feeling that the generated warning is not totally correct as its already including 24.11 information.


```
warning: Git tree '/home/shawn/dev/nix-configuration' is dirty
trace: warning: A legacy Nextcloud install (from before NixOS 24.11) may be installed.

After nextcloud28 is installed successfully, you can safely upgrade
to 29. The latest version available is Nextcloud29.

Please note that Nextcloud doesn't support upgrades across multiple major versions
(i.e. an upgrade from 16 is possible to 17, but not 16 to 18).

The package can be upgraded by explicitly declaring the service-option
`services.nextcloud.package`.
```

Be aware of the 24.11 in the warning.

Is that intended or a bug?

**Pull request merged**